src/utils.py

```diff
@@ -1065,6 +1067,19 @@ def __call__(self, atoms: Collection[GroundAtom]) -> float:
         raise NotImplementedError("Override me!")
 
 
+class GoalCountHeuristic(_TaskPlanningHeuristic):
+    """The number of goal atoms that are not in the current state."""
+    HEURISTIC_NAME: str = "goal_count"
+
+    @functools.cached_property
+    def _goal_set(self) -> Set[GroundAtom]:
+        """Convert the goal atom Collection into an explicit Set."""
```

Comment:
goals should already be `Set[GroundAtom]` (see the definition of `Task` in `structs.py` for example). did you run into a type checking issue without this? if so, we should probably change whichever Collection is causing the issue.

Reply:
```
@dataclass(frozen=True)
class _TaskPlanningHeuristic:
    """A task planning heuristic."""
    name: str
    init_atoms: Collection[GroundAtom]
    goal: Collection[GroundAtom]
    ...
```

It's still a `Collection[GroundAtom]` in the parent class which is why I added this here. That being said, `goal: Set[GroundAtom]` in `create_task_planning_heuristic` so we could change the typing to be a `Set` rather than a `Collection`

Reply:
ah I see, yeah that was our mistake. would you mind changing it there to be a Set? thanks!

Reply:
yeah, there are similarly two instances of `git grep "init_atoms: Colle"`, we should fix both as well!